File: 프로그래머스/unrated/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.py
def solution(my_string):
    answer = 0
    i=0
    while i < len(my_string):
        num = 0
        flag = 0
        while i < len(my_string) and my_string[i].isdigit() == True:
            num *= 10
            num += int(my_string[i])
            i+=1
            flag = 1
        if flag == 0:
            i += 1
        answer += num
    return answer

File: 프로그래머스/0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.java
class Solution {
    public int solution(String my_string) {
        int answer = 0;
        
        for(int i=0;i<my_string.length();i++){
            int num = 0;
            char ch = my_string.charAt(i);
            while(ch >= '0' && ch <= '9'){
                num *=10;
                num += Integer.parseInt(String.valueOf(ch));
                i++;
                if(i>=my_string.length())
                    break;
                ch = my_string.charAt(i);
            }
            answer += num;
        }
        
        return answer;
    }
}